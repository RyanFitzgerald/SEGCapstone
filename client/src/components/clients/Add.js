import React from 'react';

class Add extends React.Component {
  componentDidMount() {
    this.props.setActiveSubtab(2);
  }

  render() {
    return (
      <div className="content">
        <div className="row">
          <div className="column">
            <h2 className="card-title">Add a New Client</h2>
            <div className="card">
                <div className="row form-section">
                  <div className="md-4 column form-section__title no-left">
                    <h3>Basic Information</h3>
                    <p>
                      Enter all the basic information about this client to help you identify them later
                    </p>
                  </div>
                  <div className="md-8 column no-right">
                    <label className="form-label" htmlFor="first-name">First Name <span className="form-required">*</span></label>
                    <input name="first-name" className="form-text form-text--full" type="text"/>
                    <label className="form-label" htmlFor="last-name">Last Name <span className="form-required">*</span></label>
                    <input name="last-name" className="form-text form-text--full" type="text"/>
                    <label className="form-label" htmlFor="salesman">Sold by <span className="form-required">*</span></label>
                    <span className="form-select">
                      <select name="salesman">
                        <option>John Doe</option>
                      </select>
                    </span>
                  </div>
                </div>
                <div className="row form-section">
                    <div className="md-4 column form-section__title no-left">
                        <h3>Contact Information</h3>
                        <p>
                            Enter all the contact information about this client
                        </p>
                    </div>
                    <div className="md-8 column no-right">
                        <label className="form-label" htmlFor="email">Email <span className="form-required">*</span></label>
                        <input name="email" className="form-text form-text--full" type="text"/>
                        <label className="form-label" htmlFor="phone">Phone Number <span className="form-required">*</span></label>
                        <input name="phone" className="form-text form-text--full" type="text"/>
                    </div>
                </div>
              <div className="row form-section no-border">
                <div className="md-4 column form-section__title no-left">
                    <h3>Client Location</h3>
                    <p>
                        Enter all the information associated with the location of this client
                    </p>
                </div>
                <div className="md-8 column no-right">
                  <label className="form-label" htmlFor="street">Street <span className="form-required">*</span></label>
                  <input name="street" className="form-text form-text--full" type="text"/>
                  <label className="form-label" htmlFor="postal-code">Postal Code <span className="form-required">*</span></label>
                  <input name="postal-code" className="form-text form-text--full" type="text"/>
                  <label className="form-label" htmlFor="city">City <span className="form-required">*</span></label>
                  <span className="form-select">
                    <select name="city">
                      <option>Ottawa</option>
                    </select>
                  </span>
                </div>
              </div>
              <div className="text-center">
                  <input type="submit" className="btn btn--primary btn--large" value="Add Client"/>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Add;