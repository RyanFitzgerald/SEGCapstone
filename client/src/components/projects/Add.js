import React from 'react';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import * as api from '../../api';
import TagSelector from 'tagselector';
import Pikaday from 'pikaday';
import '../../../node_modules/pikaday/css/pikaday.css';

class Add extends React.Component {
  constructor() {
    super();

    // Bind functions
    this.handleSubmit = this.handleSubmit.bind(this);
    this.addProject = this.addProject.bind(this);

    // Set state
    this.state = {
      redirect: false
    }
  }

  componentDidMount() {
    // Set title
    document.title = 'Add Project | Renovaction';

    // Update tab
    this.props.setActiveSubtab(2);

    // Create tagselector
    let projectType = new TagSelector('project-type');
    let projectSubType = new TagSelector('project-subtype');

    // Create pikaday
    let soldDate = new Pikaday({field: document.getElementById('sold-date'), format: 'MMMM DD, YYYY'});
    let cashinDate = new Pikaday({field: document.getElementById('cashin-date'), format: 'MMMM DD, YYYY'});
    let startDate = new Pikaday({field: document.getElementById('start-date'), format: 'MMMM DD, YYYY'});
    let endDate = new Pikaday({field: document.getElementById('end-date'), format: 'MMMM DD, YYYY'});
  }

  handleSubmit(e) {
    e.preventDefault();

    let types = [];
    for (let i = 0; i < this.type.options.length; i++) {
      if (this.type.options[i].selected) {
        types.push(this.type.options[i].value);
      }
    }

    // Get form data
    const project = {
      name: this.name.value,
      street: this.street.value,
      postalCode: this.postalCode.value,
      city: this.city.value,
      soldDate: this.soldDate.value,
      startDate: this.startDate.value,
      endDate: this.endDate.value,
      cashinDate: this.cashinDate.value,
      labourCost: this.labourCost.value,
      materialsCost: this.materialsCost.value,
      actualCost: this.actualCost.value,
      status: this.status.value,
      type: types,
      client: this.client.value
    };

    // Call api
    this.addProject(project);
  }

  addProject(project) {
    api.addProject(project).then(resp => {
      // Append id
      project._id = resp;

      // Update parent state
      //this.props.addToProjects(project);

      // Redirect
      this.setState({
        redirect: `/projects/${resp}`
      });
    });
  }

  render() {
    // Define variables
    const types = this.props.types || [];
    const clients = this.props.clients || [];

    if (!this.state.redirect) {
      return (
        <div className="content">
          <div className="row">
            <div className="column">
              <h2 className="card-title">Add a New Project</h2>
              <div className="card">
                <form onSubmit={this.handleSubmit}>
                  <div className="row form-section">
                    <div className="md-4 column form-section__title no-left">
                      <h3>Basic Information</h3>
                      <p>
                        Enter all the basic information about this project to help you identify it later
                      </p>
                    </div>
                    <div className="md-8 column no-right">
                      <label className="form-label" htmlFor="name">Project Nickname <span className="form-required">*</span></label>
                      <input name="name" ref={input => this.name = input} className="form-text form-text--full" type="text" placeholder="E.g. Doe Roofing Project" required/>
                      <label className="form-label" htmlFor="client">Project Client <span className="form-required">*</span></label>
                      <span className="form-select">
                        <select name="client" ref={input => this.client = input} required>
                        {clients.map((client, key) => {
                          return <option key={key} value={client._id}>{client.name}</option>;
                        })}
                        </select>
                      </span>
                      <label className="form-label" htmlFor="client">Project Status <span className="form-required">*</span></label>
                      <span className="form-select">
                        <select name="client" ref={input => this.status = input} required>
                          <option value="Not Started">Not Started</option>
                          <option value="In Progress">In Progress</option>
                          <option value="Complete">Complete</option>
                        </select>
                      </span>
                    </div>
                  </div>
                  <div className="row form-section">
                    <div className="md-4 column form-section__title no-left">
                      <h3>Project Location</h3>
                      <p>
                        Enter all the information associated with the location of this project
                      </p>
                    </div>
                    <div className="md-8 column no-right">
                      <label className="form-label" htmlFor="street">Street <span className="form-required">*</span></label>
                      <input name="street" ref={input => this.street = input} className="form-text form-text--full" type="text" required/>
                      <label className="form-label" htmlFor="postal-code">Postal Code <span className="form-required">*</span></label>
                      <input name="postal-code" ref={input => this.postalCode = input} className="form-text form-text--full capitalize" maxLength="6" type="text" required/>
                      <label className="form-label" htmlFor="city">City <span className="form-required">*</span></label>
                      <span className="form-select" required>
                        <select name="city" ref={input => this.city = input}>
                          <option value="Ottawa">Ottawa</option>
                        </select>
                      </span>
                    </div>
                  </div>
                  <div className="row form-section">
                    <div className="md-4 column form-section__title no-left">
                      <h3>Project Type</h3>
                      <p>
                        Enter all information about this project's type and associated subtypes (if applicable)
                      </p>
                    </div>
                    <div className="md-8 column no-right">
                      <label className="form-label" htmlFor="type">Main Type <span className="form-required">*</span></label>
                      <select name="type" id="project-type" ref={input => this.type = input} required>
                        {types.map((type, key) => {
                          return <option key={key} value={type._id}>{type.name}</option>;
                        })}
                      </select>
                      <label className="form-label" htmlFor="subtype">Sub Type</label>
                      <select name="subtype" id="project-subtype">
                        <option value="1">Kitchen</option>
                        <option value="2">Bathroom</option>
                        <option value="3">Baseroom</option>
                        <option value="4">Flooring</option>
                        <option value="5">Miscellaneous</option>                
                      </select>
                    </div>
                  </div>
                  <div className="row form-section">
                    <div className="md-4 column form-section__title no-left">
                      <h3>Project Dates</h3>
                      <p>
                        Enter all known date information associated with this project (Format: January 1, 2017)
                      </p>
                    </div>
                    <div className="md-8 column no-right">
                      <label className="form-label" htmlFor="sold-date">Sold Date <span className="form-required">*</span></label>
                      <input name="sold-date" ref={input => this.soldDate = input} id="sold-date" className="form-text form-text--full" type="text" required/>
                      <label className="form-label" htmlFor="cashin-date">Cashin Date</label>
                      <input name="cashin-date" ref={input => this.cashinDate = input} id="cashin-date" className="form-text form-text--full" type="text"/>
                      <label className="form-label" htmlFor="start-date">Project Start Date</label>
                      <input name="start-date" ref={input => this.startDate = input} id="start-date" className="form-text form-text--full" type="text"/>
                      <label className="form-label" htmlFor="end-date">Project End Date</label>
                      <input name="end-date" ref={input => this.endDate = input} id="end-date" className="form-text form-text--full" type="text"/>
                    </div>
                  </div>
                  <div className="row form-section no-border">
                    <div className="md-4 column form-section__title no-left">
                      <h3>Project Costs</h3>
                      <p>
                        Enter all known cost information associated with this project
                      </p>
                    </div>
                    <div className="md-8 column no-right">
                      <label className="form-label" htmlFor="labour-cost">Labour Cost</label>
                      <input name="labour-cost" ref={input => this.labourCost = input} className="form-text form-text--full" type="text"/>
                      <label className="form-label" htmlFor="materials-cost">Materials Cost</label>
                      <input name="materials-cost" ref={input => this.materialsCost = input} className="form-text form-text--full" type="text"/>
                      <label className="form-label" htmlFor="actual-cost">Actual Cost</label>
                      <input name="actual-cost" ref={input => this.actualCost = input} className="form-text form-text--full" type="text"/>
                    </div>
                  </div>
                  <div className="text-center">
                    <input type="submit" className="btn btn--primary btn--large" value="Add Project" />
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      );
    }

    return (
      <Redirect to={this.state.redirect}/>
    );
    
  }
}

Add.propTypes = {
  setActiveSubtab: PropTypes.func.isRequired,
  types: PropTypes.array.isRequired
}

export default Add;