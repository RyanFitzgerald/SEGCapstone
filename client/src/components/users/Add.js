import React from 'react';
import Logo from '../../logo.png';
import { Redirect } from 'react-router-dom';
import * as api from '../../api';

import Loading from '../Loading';

class Add extends React.Component {
  constructor() {
    super();

    // Bind functions
		this.handleSubmit = this.handleSubmit.bind(this);
		this.addUser = this.addUser.bind(this);

    // Set state
    this.state = {
      redirect: false,
      formError: false
    }
  }

  componentDidMount() {
    // Set title
    document.title = 'Add User | Renovaction';

    // Update active tab
    this.props.setActiveSubtab(3);
  }

  handleSubmit(e) {
    // Stop form submission
    e.preventDefault();

    // Get form data
    const user = {
			email: this.email.value,
			name: this.name.value,
			password: this.password.value,
			role: this.role.value,
      access_token: JSON.parse(sessionStorage.getItem('user')).access_token
		};
		
		this.addUser(user);
  }

  addUser(user) {
    api.addUser(user).then(resp => {
      if (resp.status === 500) {
        this.setState({
          formError: 'There was an error when submitting the form, please try again.'
        })
        return;
      }

      // Append id
      user._id = resp;

      // Update parent state
      this.props.addToUsers(user);
      
      // Redirect
      this.setState({
        redirect: {
          location: `/settings/users/${resp}`,
          message: 'Successfully added user!',
          type: 'success'
        }
      });
    });
  }

  render() {
    const roles = this.props.roles || [];

		if (this.state.redirect) {
      this.props.addNotification(this.state.redirect.message, this.state.redirect.type);
      return (
        <Redirect to={this.state.redirect.location}/>
      );
    }

    return (
      <div className="content">
        <div className="row">
          <div className="column">
            <h2 className="card-title">Add New User</h2>
            <div className="card">
              {this.props.renderError(this.state.formError)}
              <form onSubmit={this.handleSubmit}>
                <div className="row">
                  <div className="md-4 column form-section__title no-left">
                    <h3>Basic Information</h3>
                    <p>
                      Enter all the basic information about this user
                    </p>
                  </div>
                  <div className="md-8 column no-right">
                    <label className="form-label" htmlFor="name"> Name <span className="form-required">*</span></label>
                    <input ref={input => this.name = input} name="name" className="form-text form-text--full" type="text" placeholder="Name" required />
                    <label className="form-label" htmlFor="email">Email <span className="form-required">*</span></label>
                    <input ref={input => this.email = input} name="email" className="form-text form-text--full" type="email" placeholder="Email" required/>
                    <label className="form-label" htmlFor="password">Password <span className="form-required">*</span></label>
                    <input ref={input => this.password = input} name="password" className="form-text form-text--full" type="password" placeholder="Password" required/>
                    <label className="form-label" htmlFor="role">Role <span className="form-required">*</span></label>
                    <span className="form-select" required>
                      <select name="role" ref={input => this.role = input}>
                      {roles.map((role, key) => {
                        return <option key={key} value={role._id}>{role.name}</option>;
                      })}
                      </select>
                    </span>
                  </div>
                </div>
                <div className="text-center">
                    <input type="submit" className="btn btn--primary btn--large" value="Add User"/>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Add;