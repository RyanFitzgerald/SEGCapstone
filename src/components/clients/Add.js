import React, { Component, PropTypes } from 'react';

class Add extends Component {

    componentDidMount() {
        // Update page title
        document.title = 'Add New Client';

        // Update Active Tab
        this.props.setHeaderTab(3);
    }

    render() {
        return (
            <div id="dashboard-other">
                <div className="row">
                    <div className="small-12 columns">
                        <h2>Add a New Client</h2>
                        <div id="client-add" className="dashboard-block">
                            <div className="row">
                                <div className="small-12 medium-6 columns">
                                    <label htmlFor="client-first-name">First Name <span className="required">*</span></label>
                                    <input type="text" name="fName" className="client-text-form" id="client-first-name" required />
                                </div>
                                <div className="small-12 medium-6 columns">
                                    <label htmlFor="client-last-name">Last Name <span className="required">*</span></label>
                                    <input type="text" name="lName" className="client-text-form" id="client-last-name" required />
                                </div>
                            </div>
                            <div className="row">
                                <div className="small-12 medium-6 columns">
                                    <label htmlFor="client-email">Email</label>
                                    <input type="email" name="email" className="client-text-form" id="client-email" />
                                </div>
                                <div className="small-12 medium-6 columns">
                                    <label htmlFor="client-phone">Phone Number <span className="required">*</span></label>
                                    <input type="text" name="phone" className="client-text-form" id="client-phone" required />
                                </div>
                            </div>
                            <div className="row">
                                <div className="small-12 medium-4 columns">
                                    <label htmlFor="client-street">Street <span className="required">*</span></label>
                                    <input type="text" name="street" className="client-text-form" id="client-street" required />
                                </div>
                                <div className="small-12 medium-4 columns">
                                    <label htmlFor="client-postal-code">Postal Code <span className="required">*</span></label>
                                    <input type="text" name="postalCode" className="client-text-form" id="client-postal-code" required />
                                </div>
                                <div className="small-12 medium-4 columns">
                                    <label htmlFor="client-city">City <span className="required">*</span></label>
                                    <select name="city" className="client-select-form" id="client-city" required>
                                        <option value="Ottawa">Ottawa</option>
                                    </select>
                                </div>
                            </div>
                            <p className="required-info">Fields marked with * are required.</p>
                            <input type="submit" name="submit" className="btn" id="client-submit" value="Add Client" />
                        </div>
                        {/* <-- End #client-add -->*/}
                    </div>
                </div>
            </div>
        );
    }
}

Add.propTypes = {
    setHeaderTab: PropTypes.func.isRequired
};

export default Add;
