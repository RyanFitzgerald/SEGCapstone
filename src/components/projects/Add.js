import React, { Component, PropTypes } from 'react';

class Add extends Component {

    componentDidMount() {
        // Update page title
        document.title = 'Add New Client';

        // Update Active Tab
        this.props.setHeaderTab(3);
    }

    render() {
        return (
            <div id="dashboard-other">
                <div className="row">
                    <div className="small-12 columns">
                        <h2>Add a New Project</h2>
                        <div id="project-add" className="dashboard-block">
                            <div className="row">
                                <div className="small-12 columns">
                                    <label htmlFor="project-name">Name <span className="required">*</span></label>
                                    <input type="text" name="fName" className="project-text-form" id="project-name" required />
                                </div>
                            </div>
                            <div className="row">
                                <div className="small-12 medium-6 columns">
                                    <label htmlFor="project-type">Type  <span className="required">*</span></label>
                                    <select name="type" className="project-select-form" id="project-type" multiple required>
                                        <option value="Ottawa">Roofing</option>
                                        <option value="Ottawa">Siding</option>
                                        <option value="Ottawa">Other</option>
                                    </select>
                                </div>
                                <div className="small-12 medium-6 columns">
                                    <label htmlFor="project-description">Description</label>
                                    <textarea name="description" className="project-textarea-form" id="project-description"></textarea>
                                </div>
                            </div>
                            <div className="row">
                                <div className="small-12 medium-4 columns">
                                    <label htmlFor="project-street">Street <span className="required">*</span></label>
                                    <input type="text" name="street" className="project-text-form" id="project-street" required />
                                </div>
                                <div className="small-12 medium-4 columns">
                                    <label htmlFor="project-postal-code">Postal Code <span className="required">*</span></label>
                                    <input type="text" name="postalCode" className="project-text-form" id="project-postal-code" required />
                                </div>
                                <div className="small-12 medium-4 columns">
                                    <label htmlFor="project-city">City <span className="required">*</span></label>
                                    <select name="city" className="project-select-form" id="project-city" required>
                                        <option value="Ottawa">Ottawa</option>
                                    </select>
                                </div>
                            </div>
                            <div className="row">
                                <div className="small-12 medium-6 columns">
                                    <label htmlFor="project-start">Start Date</label>
                                    <input type="date" name="startDate" className="project-text-form" id="project-start" />
                                </div>
                                <div className="small-12 medium-6 columns">
                                    <label htmlFor="project-end">End Date</label>
                                    <input type="date" name="endDate" className="project-text-form" id="project-end" />
                                </div>
                            </div>
                            <div className="row">
                                <div className="small-12 medium-6 columns">
                                    <label htmlFor="project-estimate">Estimated Cost ($)</label>
                                    <input type="text" name="estimatedCost" className="project-text-form" id="project-estimate" />
                                </div>
                                <div className="small-12 medium-6 columns">
                                    <label htmlFor="project-cost">Actual Cost ($)</label>
                                    <input type="text" name="actualCost" className="project-text-form" id="project-cost" />
                                </div>
                            </div>
                            <div className="row">
                                <div className="small-12 columns">
                                    <label htmlFor="project-contract">Contract</label>
                                    <input type="file" name="contract" className="project-upload-form" id="project-contract" />
                                </div>
                            </div>
                            <div className="row">
                                <div className="small-12 columns">
                                    <label htmlFor="project-client">Project Client <span className="required">*</span></label>
                                    <input type="text" name="estimatedCost" className="project-text-form" id="project-client" required />
                                </div>
                            </div>
                            <p className="required-info">Fields marked with * are required.</p>
                            <input type="submit" name="submit" className="btn" id="project-submit" value="Add Project" />
                        </div>
                        {/* <-- End #client-add --> */}
                    </div>
                </div>
            </div>
        );
    }
}

Add.propTypes = {
    setHeaderTab: PropTypes.func.isRequired
};

export default Add;
