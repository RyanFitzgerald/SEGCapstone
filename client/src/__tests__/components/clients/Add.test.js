import React from 'react';
import Add from '../../../Components/clients/Add';

describe('<Add />', () => {
  it('should render correctly', () => {   
    // Create mock functions / spies
    const setActiveSubtabMock = jest.fn();
    const renderErrorMock = jest.fn();
    
    // Shallow render it
    const wrapper = shallow(
      <Add setActiveSubtab={setActiveSubtabMock} renderError={renderErrorMock}/>
    );

    // Compare to old snapshot
    expect(shallowToJson(wrapper)).toMatchSnapshot();

    // Check set active subtab is called and with correct value
    expect(setActiveSubtabMock).toHaveBeenCalledTimes(1);
    expect(setActiveSubtabMock.mock.calls[0][0]).toBe(2);
  });
});

