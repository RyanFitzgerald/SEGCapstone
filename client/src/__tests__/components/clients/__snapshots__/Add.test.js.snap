// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Add /> should render correctly 1`] = `
<div
  className="content"
>
  <div
    className="row"
  >
    <div
      className="column"
    >
      <h2
        className="card-title"
      >
        Add a New Client
      </h2>
      <div
        className="card"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="row form-section"
          >
            <div
              className="md-4 column form-section__title no-left"
            >
              <h3>
                Basic Information
              </h3>
              <p>
                Enter all the basic information about this client to help you identify them later
              </p>
            </div>
            <div
              className="md-8 column no-right"
            >
              <label
                className="form-label"
                htmlFor="firstName"
              >
                 First Name 
                <span
                  className="form-required"
                >
                  *
                </span>
              </label>
              <input
                className="form-text form-text--full"
                name="firstName"
                required={true}
                type="text"
              />
              <label
                className="form-label"
                htmlFor="lastName"
              >
                 Last Name 
                <span
                  className="form-required"
                >
                  *
                </span>
              </label>
              <input
                className="form-text form-text--full"
                name="lastName"
                required={true}
                type="text"
              />
              <label
                className="form-label"
                htmlFor="salesman"
              >
                Sold by 
                <span
                  className="form-required"
                >
                  *
                </span>
              </label>
              <span
                className="form-select"
              >
                <select
                  name="salesman"
                  required={true}
                >
                  <option>
                    John Doe
                  </option>
                </select>
              </span>
            </div>
          </div>
          <div
            className="row form-section"
          >
            <div
              className="md-4 column form-section__title no-left"
            >
              <h3>
                Contact Information
              </h3>
              <p>
                Enter all the contact information about this client
              </p>
            </div>
            <div
              className="md-8 column no-right"
            >
              <label
                className="form-label"
                htmlFor="email"
              >
                Email
              </label>
              <input
                className="form-text form-text--full"
                name="email"
                type="email"
              />
              <label
                className="form-label"
                htmlFor="homePhone"
              >
                Home Phone
              </label>
              <input
                className="form-text form-text--full"
                maxLength="12"
                name="homePhone"
                onKeyUp={[Function]}
                type="text"
              />
              <label
                className="form-label"
                htmlFor="mobilePhone"
              >
                Mobile Phone
              </label>
              <input
                className="form-text form-text--full"
                maxLength="12"
                name="mobilePhone"
                onKeyUp={[Function]}
                type="text"
              />
              <label
                className="form-label"
                htmlFor="workPhone"
              >
                Work Phone
              </label>
              <input
                className="form-text form-text--full"
                maxLength="12"
                name="workPhone"
                onKeyUp={[Function]}
                type="text"
              />
            </div>
          </div>
          <div
            className="row form-section no-border"
          >
            <div
              className="md-4 column form-section__title no-left"
            >
              <h3>
                Client Location
              </h3>
              <p>
                Enter all the information associated with the location of this client
              </p>
            </div>
            <div
              className="md-8 column no-right"
            >
              <label
                className="form-label"
                htmlFor="houseNumber"
              >
                House Number 
                <span
                  className="form-required"
                >
                  *
                </span>
              </label>
              <input
                className="form-text form-text--full"
                name="houseNumber"
                required={true}
                type="text"
              />
              <label
                className="form-label"
                htmlFor="street"
              >
                Street 
                <span
                  className="form-required"
                >
                  *
                </span>
              </label>
              <input
                className="form-text form-text--full"
                name="street"
                required={true}
                type="text"
              />
              <label
                className="form-label"
                htmlFor="city"
              >
                City 
                <span
                  className="form-required"
                >
                  *
                </span>
              </label>
              <input
                className="form-text form-text--full"
                name="city"
                required={true}
                type="text"
              />
              <label
                className="form-label"
                htmlFor="postal-code"
              >
                Postal Code 
                <span
                  className="form-required"
                >
                  *
                </span>
              </label>
              <input
                className="form-text form-text--full capitalize"
                maxLength="6"
                name="postal-code"
                required={true}
                type="text"
              />
            </div>
          </div>
          <div
            className="text-center"
          >
            <input
              className="btn btn--primary btn--large"
              type="submit"
              value="Add Client"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
